src/braket/circuits/noise.py

```diff
@@ -457,58 +466,63 @@ def __init__(
         Raises:
             ValueError: If the `qubit_count` is less than 1, `ascii_symbols` are `None`, or
                 `ascii_symbols` length != `qubit_count`, `probX` or `probY` or `probZ`
-                is not `float` or FreeParameter, `probX` or `probY` or `probZ` > 1.0, or
+                is not `float` or FreeParameterExpression, `probX` or `probY` or `probZ` > 1.0, or
                 `probX` or `probY` or `probZ` < 0.0, or `probX`+`probY`+`probZ` > 1
         """
         super().__init__(qubit_count=qubit_count, ascii_symbols=ascii_symbols)
 
         total = 0
-        self._parameters = []
-        total += self._add_param("probX", probX)
-        total += self._add_param("probY", probY)
-        total += self._add_param("probZ", probZ)
+        total += PauliNoise._validate_param("probX", probX)
+        total += PauliNoise._validate_param("probY", probY)
+        total += PauliNoise._validate_param("probZ", probZ)
         if total > 1:
             raise ValueError("the sum of probX, probY, probZ cannot be larger than 1")
+        self._parameters = [probX, probY, probZ]
 
-    def _add_param(self, paramName, param: Union[FreeParameter, float]):
-        if isinstance(param, FreeParameter):
-            self._parameters.append(param)
+    @staticmethod
+    def _validate_param(param_name, param: Union[FreeParameterExpression, float]):
```

Comment:
Can you add a return type hint, and some documentation? It's not clear why a validation function is returning a float.